yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -476,14 +509,17 @@ export class Archiver implements ArchiveSource {
 
   /**
    * Gets an l2 block.
-   * @param number - The block number to return (inclusive).
+   * @param number - The block number to return.
    * @returns The requested L2 block.
    */
   public async getBlock(number: number): Promise<L2Block | undefined> {
     // If the number provided is -ve, then return the latest block.
     if (number < 0) {
       number = await this.store.getSynchedL2BlockNumber();
     }
+    if (number == 0) {
```

Comment:
Isn't this genesis block? Should it be undefined?

Reply:
There is a "genesis" state, but there is not actually a block that you have.